dlib/vectorstream/vectorstream.h

```diff
@@ -120,14 +145,15 @@ namespace dlib
             std::iostream(&buf),
             buf(buffer)
         {}
-
-        std::istream& seekg (
-            std::streampos pos
-        )
-        {
-            buf.seekg(pos);
-            return *this;
-        }
+            
+        vectorstream(const vectorstream& ori) = delete;
+            
+        vectorstream(
+            vectorstream&& item
+        ) : 
+            std::iostream(std::move(item)), 
+            buf(std::move(item.buf))
+        {}
```

Comment:
Why add this?  This seems to be asking to break user code since this class doesn't own the vector buffer.  This change would allow someone to make a vectorstream in a function and then return it.  In such a case it is very likely they created the vectorstream referencing a local buffer that is no defunct.

Reply:
I initially needed the move constructor of vectorstream for proxy_serialize to work (the way i initially implemented it). Now, using a unique_ptr, it's not needed. Yep you're right. So both the copy constructor and the move constructor should be marked deleted. Or just leave it as it was, i.e. default implementation

Reply:
Given your argument, it sounds like explicitly deleting both copy and move constructors is the safest thing to do.